x-pack/plugins/index_management/public/application/components/mappings_editor/components/document_fields/field_parameters/select_inference_id.tsx

```diff
@@ -136,40 +163,48 @@ export const SelectInferenceId = ({
     };
     setOptions(Object.values(mergedOptions));
   }, [inferenceIdOptionsFromModels, defaultInferenceIds]);
-  const [isCreateInferenceApiLoading, setIsCreateInferenceApiLoading] = useState(false);
+
+  const { toasts } = useComponentTemplatesContext();
 
   const onSaveInferenceCallback = useCallback(
     async (inferenceId: string, taskType: InferenceTaskType, modelConfig: ModelConfig) => {
-      setIsCreateInferenceApiLoading(true);
+      setIsInferenceFlyoutVisible(!isInferenceFlyoutVisible);
+      const oldOptions = options;
       try {
-        await ml?.mlApi?.inferenceModels?.createInferenceEndpoint(
-          inferenceId,
-          taskType,
-          modelConfig
-        );
-        setIsInferenceFlyoutVisible(!isInferenceFlyoutVisible);
-        setIsCreateInferenceApiLoading(false);
-        setInferenceAddError(undefined);
+        const isDeployable =
+          modelConfig.service === Service.elser || modelConfig.service === Service.elasticsearch;
+        if (isDeployable) NotificationToasts({ toasts });
+
+        const combined: EuiSelectableOption[] = [
+          {
+            label: inferenceId,
+            'data-test-subj': `custom-inference_${inferenceId}`,
+          },
+        ];
+        setOptions([...oldOptions, ...combined]);
+
         const trainedModelStats = await ml?.mlApi?.trainedModels.getTrainedModelStats();
         const defaultEndpointId =
           inferenceServiceTypeElasticsearchModelMap[modelConfig.service] || '';
         const newModelId: InferenceToModelIdMap = {};
         newModelId[inferenceId] = {
           trainedModelId: defaultEndpointId,
-          isDeployable:
-            modelConfig.service === Service.elser || modelConfig.service === Service.elasticsearch,
-          isDeployed: getTrainedModelStats(trainedModelStats)[defaultEndpointId] === 'deployed',
-          defaultInferenceEndpoint: false,
+          isDeployable,
+          isDeployed:
+            getTrainedModelStats(trainedModelStats)[defaultEndpointId] === DeploymentState.DEPLOYED,
         };
-        resendRequest();
-        setNewInferenceEndpoint(newModelId);
+        const customInferenceEndpointConfig: CustomInferenceEndpointConfig = {
+          taskType,
+          modelConfig,
+        };
+        setNewInferenceEndpoint(newModelId, customInferenceEndpointConfig);
       } catch (error) {
-        const errorObj = extractErrorProperties(error);
-        setInferenceAddError(errorObj.message);
-        setIsCreateInferenceApiLoading(false);
+        NotificationToasts({ toasts, error });
```

Comment:
There are multiple places where we show the same success and error toasts message. So thought of creating a component with the success & error message, so that we don't have repeat the code block again but rather can just call the component. 

Are you suggesting that we that we shouldn't have a separate component to deal with toasts but rather write them out?

Reply:
You can have functions like `showSuccess` and `showError` that you can call in these specific spots. 

The issue is that you're embedding this one component in every spot, and then having the component check which function it should add to the toasts service. But that's superfluous: you're adding an extra conditional check that's unnecessary. If you're in this spot in the code, you know it's always going to be an error toast. We don't need to have the component determine that for us, we *know* it. 

So we can have a shared function for error toasts and a shared function for success toasts, but those two don't need to be combined into one parent component. They can just live on their own.

Reply:
Thank you for explaining this in detail.  I agree, having a function is more neat and organized. 

I have divided this into two functions `showMlSuccessToasts` & `showMlErrorToasts`, declared within a parent component [MLModelNotificationToasts](https://github.com/elastic/kibana/blob/3b73d9485ca7b9b959aa5896ae3f0f1d95796359/x-pack/plugins/index_management/public/application/components/mappings_editor/components/document_fields/ml_model_toasts.ts#L11C17-L11C42) , to avoid breaking react hook rules.